src/theory/rewriter.cpp

```diff
@@ -195,7 +203,7 @@ Node Rewriter::rewriteTo(theory::TheoryId theoryId,
 
   // Check if it's been cached already
   Node cached = getPostRewriteCache(theoryId, node);
-  if (!cached.isNull() && (tcpg == nullptr || tcpg->hasRewriteStep(node)))
+  if (!cached.isNull() && (tcpg == nullptr || node.getAttribute(rpfa)))
```

Comment:
why is it done like this now?

Reply:
This was actually incorrect if we registered a rewrite for a term without registering the proofs for its children.  We should only mark the attribute after the post-traversal of a term.